### PR TITLE
P@k ranking

### PR DESCRIPTION
**Pull request by Eleven1Liu**
Move all `P@k`, `R@k` into `precision_recall_at_ks`.

### CODE REVIEW

File: metrics.py [hunk 1]
```diff
@@ -13,30 +13,12 @@ def macro_f1(y_true, y_pred):
...
-
-    return np.mean(vals)
+def precision_recall_at_ks(y_true, y_pred_vals, top_ks):
+    rank_mat = np.argsort(-y_pred_vals)
+    denom = y_true.sum(axis=1) + 1e-10
+    scores = {}
+    for k in top_ks:
+        y_pred = np.take_along_axis(y_true, rank_mat[:,:k], axis=1)
+        scores[f'P@{k}'] = np.mean(np.sum(y_pred, axis=1) / k).item()  # precision at k
+        scores[f'R@{k}'] = np.mean(y_pred.sum(axis=1) / denom).item()  # recall at k
```
Comment (sian-chen) [inline metrics.py:23]: Improve readability.
```suggestion
    y_pred_rank = np.argsort(-y_pred_vals)
    n_pos = y_true.sum(axis=1)
    scores = {}
    for k in top_ks:
        n_pos_in_top_k = np.take_along_axis(y_true, y_pred_rank[:,:k], axis=1).sum(axis=1)
        scores[f'P@{k}'] = np.mean(n_pos_in_top_k / k).item()  # precision at k
        scores[f'R@{k}'] = np.mean(n_pos_in_top_k / (n_pos + 1e-10)).item()  # recall at k
```

Comment (cjlin1) [reply metrics.py:23]: y_pred_rank isn't very clear as it's not really ranks. Maybe y_pred_ranked_dataID.
A bit long but it's used only in line 21

Comment (Eleven1Liu) [reply metrics.py]: y_pred_ranked_idx?

**Comment by cjlin1** [comment]
Looks good

**Review by sian-chen — APPROVED** [review]

**Pull request merged**